### ISSUE TITLE
Documentation Comments Style Guide - @param

### ISSUE THREAD

**Issue by gianmarcoschifone** [checkstyle/checkstyle#19919]
Parent issue: https://github.com/checkstyle/checkstyle/issues/19807

[Coverage table](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/8c24aed_20260520164406/doc_comments_style.html#Coverage_table)

[Chapter link](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#@param)
### Image

<img width="1058" height="629" alt="Image" src="https://github.com/user-attachments/assets/56959306-f71b-4523-bff8-1aa149143386" />

#### @param (reference page)
The @param tag is followed by the name (not data type) of the parameter, followed by a description of the parameter. By convention, the first noun in the description is the data type of the parameter. (Articles like "a", "an", and "the" can precede the noun.) An exception is made for the primitive int, where the data type is usually omitted. Additional spaces can be inserted between the name and description so that the descriptions line up in a block. Dashes or other punctuation should not be inserted before the description, as the Javadoc tool inserts one dash.

Parameter names are lowercase by convention. The data type starts with a lowercase letter to indicate an object rather than a class. The description begins with a lowercase letter if it is a phrase (contains no verb), or an uppercase letter if it is a sentence. End the phrase with a period only if another phrase or sentence follows it.

Example:

```
* @param ch        the character to be tested
* @param observer  the image observer to be notified
```

Do not bracket the name of the parameter after the @param tag with `<code>...</code>` since Javadoc 1.2 and later automatically do this. (Beginning with 1.4, the name cannot contain any HTML, as Javadoc compares the @param name to the name that appears in the signature and emits a warning if there is any difference.)

When writing the comments themselves, in general, start with a phrase and follow it with sentences if they are needed.

When writing a phrase, do not capitalize and do not end with a period:

```
@param x  the x-coordinate, measured in pixels
```

When writing a phrase followed by a sentence, do not capitalize the phrase, but end it with a period to distinguish it from the start of the next sentence:

```
@param x  the x-coordinate. Measured in pixels.
```

If you prefer starting with a sentence, capitalize it and end it with a period:

```
@param x  Specifies the x-coordinate, measured in pixels.
```

When writing multiple sentences, follow normal sentence rules:

```
@param x  Specifies the x-coordinate. Measured in pixels.
```

Also see [order of multiple @param tags](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#multiple@param).

## Guidelines
This chapter contains mainly two guidelines:
1. >The @param tag is followed by the name (not data type) of the parameter
2. >...followed by a description of the parameter

Rule number 1 is enforced by [JavadocMethod](https://checkstyle.sourceforge.io/checks/javadoc/javadocmethod.html#JavadocMethod).
Rule number 2 is enforced by [NonEmptyAtclauseDescription](https://checkstyle.sourceforge.io/checks/javadoc/nonemptyatclausedescription.html#NonEmptyAtclauseDescription)

<img width="585" height="179" alt="Image" src="https://github.com/user-attachments/assets/a3b5af84-61cc-42ac-a0c5-5f50c4d47820" />

**Comment by gianmarcoschifone** [comment]
@Zopsss @romani 
There is also this guideline, what do you think?:
>By convention, the first noun in the description is the data type of the parameter. (Articles like "a", "an", and "the" can precede the noun.) An exception is made for the primitive int, where the data type is usually omitted. Additional spaces can be inserted between the name and description so that the descriptions line up in a block. Dashes or other punctuation should not be inserted before the description, as the Javadoc tool inserts one dash.

It could probably be implemented as a check in some form, but maybe this is mostly a wording guideline.

**Comment by Zopsss** [comment]
The guideline you mentioned is mostly wording but the following part can be considered as a rule and implemented as a separate check:

> Dashes or other punctuation should not be inserted before the description, as the Javadoc tool inserts one dash.

Between the `@param paramName` and it's description, there should only be whitespaces. No other special characters are allowed. The description of the parameter should not have any special characters before it, it should only be preceded with whitespace(s).

Following are the valid examples:

```java
@param fullName the full name of the person (first and last name)
@param age      the person's age in years
```

```java
@param fullName the full name of the person (first and last name)
@param age the person's age in years
```

Notice that in the second example, descriptions are not vertically aligned and it is allowed as the guideline said:

> Additional spaces can be inserted between the name and description so that the descriptions line up in a block.

It specifically said `can be inserted`, this means it is not compulsory to align the descriptions. Though if we want, we can implement this rule (parameters description should be vertically aligned) as a separate check but for now I don't think there is a need for this. If some user wants this specific rule, only then we should implement it.

Following are the invalid examples:

```java
@param paramName - description...
@param paramName : description...
@param paramName — description...
```

**Comment by gianmarcoschifone** [comment]
@Zopsss ok, that makes sense.

Should I open a separate issue for a new check that validates that no punctuation is inserted between the `@param` name and its description?

**Comment by Zopsss** [comment]
Yes always open a new issue for new check
